src/SessionCipher.js

```diff
@@ -240,7 +240,10 @@ SessionCipher.prototype = {
     });
   },
   fillMessageKeys: function(chain, counter) {
-      if (Object.keys(chain.messageKeys).length >= 1000) {
+      var ourNumber = textsecure.storage.user.getNumber();
```

Comment:
I don't think this library should care about a "current user" concept, as it generally operates on a per-device level. I would just make the limit an option on SessionCipher, then the application can decide:
```
new SessionCipher(store, address); // limit 1000
new SessionCipher(store, address, isOurNumber ? { fillMessageKeysLimit: 5000 }); // limit 5000
new SessionCipher(store, address, isOurNumber ? { fillMessageKeysLimit: false }); // no limit
```